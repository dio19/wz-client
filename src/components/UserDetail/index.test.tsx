import { render, fireEvent, screen } from '@testing-library/react';
import UsersDetail from './index';
import {BrowserRouter as Router} from 'react-router-dom';

describe('Tests Users Table', () => {

  it('should match snapshot', () => {

    const props = {
        eventHandler: jest.fn(),
    };
      
    const { asFragment } = render(<Router><UsersDetail {...props}/></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test get back event handler', async () => {
    const props = {
      eventHandler: jest.fn(),
    };

    render(<Router><UsersDetail {...props}/></Router>);
    fireEvent.click(screen.getByRole('button'));
    expect(props.eventHandler).toBeCalled();
  });

});