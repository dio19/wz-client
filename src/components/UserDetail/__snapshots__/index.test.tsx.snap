// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Users Table should match snapshot 1`] = `<DocumentFragment />`;
